cli/command/container/attach.go

```diff
@@ -66,6 +66,9 @@ func runAttach(dockerCli command.Cli, opts *attachOptions) error {
 	ctx := context.Background()
 	client := dockerCli.Client()
 
+	// request channel to wait for client
+	resultC, errC := client.ContainerWait(ctx, opts.container, "")
```

Comment:
Oh, actually I think this should come after `inspectContainerAndCheckState()` which is a sanity check on the state of the container.

Reply:
I wanted to avoid a race when by the time we return from `inspectContainerAndCheckState()` the container had exited, but I don't mind moving it afterwards.

Reply:
good point, maybe it is more correct before the inspect. The error from `ContainerWait()` wouldn't be reported until later, so the user would still see the helpful error message.

I think this is correct, ignore my comment.